tests/integration/test_integration.py

```diff
@@ -67,66 +67,143 @@ async def build_charms_and_deploy_vault(ops_test: OpsTest):
     )
     await ops_test.model.wait_for_idle(
         apps=[APPLICATION_NAME],
-        status="active",
+        status="blocked",
         timeout=1000,
         wait_for_exact_units=NUM_VAULT_UNITS,
     )
 
     return {"vault-kv-requirer": vault_kv_requirer_charm}
 
 
+@pytest.fixture(scope="module")
+async def initialize_leader_vault(
+    ops_test: OpsTest, build_charms_and_deploy_vault: Dict[str, Path | str]
+) -> Tuple[int, str, str]:
+    leader_unit = await get_leader_unit(ops_test.model, APPLICATION_NAME)
+    leader_unit_index = int(leader_unit.name[-1])
```

Comment:
This doesn't seem very future proof. What if the unit isn't a single digit? Maybe split on `/` first?

Reply:
I *highly* doubt the gh runners can handle more than 10 pods, but splitting on `/` is just better so I changed it.